modules/lang-painless/src/main/java/org/elasticsearch/painless/PainlessScriptEngine.java

```diff
@@ -440,11 +440,8 @@ private CompilerSettings buildCompilerSettings(Map<String, String> params) {
         return compilerSettings;
     }
 
-    @SuppressForbidden(
-        reason = "TODO Deprecate any lenient usage of Boolean#parseBoolean https://github.com/elastic/elasticsearch/issues/128993"
-    )
     private static boolean parseBoolean(String value) {
-        return Boolean.parseBoolean(value);
+        return LenientBooleans.parseAndCheckForDeprecatedUsage(value, LenientBooleans.Category.COMPILER_OPTION);
```

Comment:
These are definitely not user-facing and should not need deprecation. Please feel free to just switch this, and if any fail then the script engine should be updated to accommodate.